### PR TITLE
Create constant for max slowmode

### PR DESCRIPTION
**Pull request by duncte123**
[contributing]: https://github.com/DV8FromTheWorld/JDA/wiki/5%29-Contributing

## Pull Request Etiquette

- [x] I have checked the PRs for upcoming features/bug fixes.
- [x] I have read the [contributing guidelines][contributing].

### Changes

- [x] Internal code
- [x] Library interface (affecting end-user code) 
- [x] Documentation
- [ ] Other: \_____ 

Closes Issue: NaN

## Description

This pull request adds a field for the maximum slowmode that a text channel can have.
I also added an extra check to the channel action class making it throw if the slowmode was not set on a text channel.

### CODE REVIEW

**Comment by caneleex** [comment]
Shouldn't constants be `public static final`?

**Comment by duncte123** [comment]
> Shouldn't constants be `public static final`?

TextChannel is an interface so this is not needed

File: src/main/java/net/dv8tion/jda/internal/requests/restaction/ChannelActionImpl.java [hunk 1]
```diff
@@ -146,7 +146,9 @@ public ChannelType getType()
     @CheckReturnValue
     public ChannelActionImpl<T> setSlowmode(int slowmode)
     {
-        Checks.check(slowmode <= 21600 && slowmode >= 0, "Slowmode must be between 0 and 21600 (seconds)!");
+        if (type != ChannelType.TEXT)
+            throw new IllegalStateException("Can only set slowmode on text channels");
```
Comment (MinnDevelopment) [inline src/main/java/net/dv8tion/jda/internal/requests/restaction/ChannelActionImpl.java]: Is this exception documented anywhere?

File: src/main/java/net/dv8tion/jda/internal/requests/restaction/ChannelActionImpl.java [hunk 2]
```diff
@@ -146,7 +146,9 @@ public ChannelType getType()
...
+        Checks.check(slowmode <= TextChannel.MAX_SLOWMODE && slowmode >= 0, "Slowmode must be between 0 and %s (seconds)!", TextChannel.MAX_SLOWMODE);
```
Comment (MinnDevelopment) [inline src/main/java/net/dv8tion/jda/internal/requests/restaction/ChannelActionImpl.java]: This should be `%d` not `%s`.

File: src/main/java/net/dv8tion/jda/internal/requests/restaction/ChannelActionImpl.java (hunk 1 above)
Comment (duncte123) [reply src/main/java/net/dv8tion/jda/internal/requests/restaction/ChannelActionImpl.java]: Here https://ci.dv8tion.net/job/JDA/javadoc/net/dv8tion/jda/api/requests/restaction/ChannelAction.html#setSlowmode(int%29

File: src/main/java/net/dv8tion/jda/api/requests/restaction/ChannelAction.java [hunk 3]
```diff
@@ -166,6 +166,8 @@
      * @param  slowmode
      *         The number of seconds required to wait between sending messages in the channel.
      *
+     * @throws UnsupportedOperationException
+     *         If this ChannelAction is not for a TextChannel
      * @throws IllegalArgumentException
      *         If the {@code slowmode} is greater than 21600, or less than 0
```
Comment (MinnDevelopment) [inline src/main/java/net/dv8tion/jda/api/requests/restaction/ChannelAction.java]: You should use an `{@link TextChannel.MAX_SLOWMODE}` here.

**Pull request merged**